src/main/java/sevensmurfs/rehub/repository/DoctorRepository.java

```diff
@@ -1,10 +1,20 @@
 package sevensmurfs.rehub.repository;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
+import sevensmurfs.rehub.enums.Role;
 import sevensmurfs.rehub.model.entity.Doctor;
+import sevensmurfs.rehub.model.entity.Room;
+import sevensmurfs.rehub.model.entity.UserRole;
+
+import java.util.List;
+import java.util.Optional;
 
 @Repository
 public interface DoctorRepository extends JpaRepository<Doctor, Long> {
 
+    @Query("SELECT d FROM Doctor d WHERE d.firstName = :firstName AND d.lastName = :lastName")

```

Comment:
@Query is not needed.
This method should return Optional<Doctor>.
findByFirstNameAndLastName name of the method would do the same thing faster and cleaner.
Optional serves that if there is no doctor you can throw exception. See examples of other repository methods,